src/hotspot/share/memory/universe.cpp

```diff
@@ -87,6 +87,33 @@
 #include "utilities/ostream.hpp"
 #include "utilities/preserveException.hpp"
 
+// A helper class for caching a Method* when the user of the cache
+// only cares about the latest version of the Method*.  This cache safely
+// interacts with the RedefineClasses API.
+class LatestMethodCache {
+  // We save the Klass* and the idnum of Method* in order to get
+  // the current cached Method*.
+ private:
+  Klass*                _klass;
+  int                   _method_idnum;
+
+ public:
+  LatestMethodCache()   { _klass = nullptr; _method_idnum = -1; }
+
+  void   init(JavaThread* current, InstanceKlass* ik, const char* method,
+              Symbol* signature, bool is_static);
+  Klass* klass() const           { return _klass; }
+  int    method_idnum() const    { return _method_idnum; }
+
+  Method* get_method();
+};
+
+static LatestMethodCache _finalizer_register_cache; // static method for registering finalizable objects
```

Comment:
Could you align these comments?

Reply:
I aligned the comments and simplified the text.